File: 1559-maximum-profit-of-operating-a-centennial-wheel/python3/main.py
class Solution:
    def minOperationsMaxProfit(
        self, customers: List[int], boardingCost: int, runningCost: int
    ) -> int:
        result, wait, maxMoney, money, i, n = -1, 0, 0, 0, 0, len(customers)
        while i < n or wait > 0:
            wait += customers[i] if i < n else 0
            current = wait if wait < 4 else 4
            wait -= current

            i += 1
            money += current * boardingCost - runningCost
            if money > maxMoney:
                maxMoney = money
                result = i

        return result

File: 1559-maximum-profit-of-operating-a-centennial-wheel/go/main.go
func minOperationsMaxProfit(customers []int, boardingCost int, runningCost int) int {
	result, wait, max, money, i, n := -1, 0, 0, 0, 0, len(customers)

	for i < n || wait > 0 {
		if i < n {
			wait += customers[i]
		}
		current := wait
		if current > 4 {
			current = 4
		}

		i++
		wait -= current
		money += current*boardingCost - runningCost
		if money > max {
			max = money
			result = i
		}
	}
	return result

}
